### PR TITLE
UKRIW-80 -  (subtask UKRIW-365) action creator nodes component

### PR DESCRIPTION
**Pull request by KatarzynaPaszko**

### CODE REVIEW

File: libs/shared/design-system/src/lib/icon/icons/action-creator-arrow.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const ActionCreatorArrow = () => (
+  <svg xmlns='http://www.w3.org/2000/svg' width='8' height='29' viewBox='0 0 8 29' fill='none'>
+    <path
+      d='M4 0.333333C2.52724 0.333333 1.33333 1.52724 1.33333 3C1.33333 4.47276 2.52724 5.66667 4 5.66667C5.47276 5.66667 6.66667 4.47276 6.66667 3C6.66667 1.52724 5.47276 0.333333 4 0.333333ZM3.64645 28.3536C3.84171 28.5488 4.15829 28.5488 4.35355 28.3536L7.53553 25.1716C7.7308 24.9763 7.7308 24.6597 7.53553 24.4645C7.34027 24.2692 7.02369 24.2692 6.82843 24.4645L4 27.2929L1.17157 24.4645C0.97631 24.2692 0.659727 24.2692 0.464465 24.4645C0.269203 24.6597 0.269203 24.9763 0.464465 25.1716L3.64645 28.3536ZM3.5 3L3.5 28L4.5 28L4.5 3L3.5 3Z'
+      fill='#4483FF'
```
Comment (KatarzynaPaszko) [inline libs/shared/design-system/src/lib/icon/icons/action-creator-arrow.tsx:5]: I added color for now, as this is temporary arrow that will be replaced by ReactFlow framework arrows

File: libs/map/feature-action-creator-panel/src/lib/content/workflow/pseudo-input.component.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+};
+
+interface IPseudoInputProps {
+  value?: string;
+  onChange?: (value: string) => void;
+  className?: string;
+  iconName?: 'Polygon' | 'Circle' | 'Satellite' | 'Square';
+}
+
+export const PseudoInput: React.FC<IPseudoInputProps> = ({ value, onChange, className, iconName = 'Satellite' }) => {
```
Comment (mlu-ssoft) [inline libs/map/feature-action-creator-panel/src/lib/content/workflow/pseudo-input.component.tsx:21]: this should be normal input. In the future we will use react hook form to validate data in Workflow tab

File: libs/shared/design-system/src/lib/icon/icons/action-creator-arrow.tsx (hunk 1 above)
Comment (mlu-ssoft) [reply libs/shared/design-system/src/lib/icon/icons/action-creator-arrow.tsx:5]: ok

File: libs/shared/design-system/src/lib/select/select.tsx [hunk 3]
```diff
@@ -1,76 +1,120 @@
...
 }
 
 export const Select = ({
   options,
   onChange,
   placeholder = 'GLOBAL.DESIGN_SYSTEM.SELECT.PLACEHOLDER',
   error,
+  className,
+  value,
 }: ISelectProps) => {
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/select/select.tsx:32]: This select should be not connected component (without internal state). This way we will be able to use it with react hook forms. If you need component with state then create two components - one basic, second one with state (eg. ConnectedSelect or something like that)

File: libs/map/feature-action-creator-panel/src/lib/content/workflow/function-input.component.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  { value: 'False colour (urban)', label: 'False colour (urban)' },
+  { value: 'Moisture index', label: 'Moisture index' },
+  { value: 'SWIR', label: 'SWIR' },
+  { value: 'NDWI', label: 'NDWI' },
+  { value: 'NDSI', label: 'NDSI' },
+];
+
+interface IFunctionSelectInputProps {
+  visible?: boolean;
+  onChange: (value: string) => void;
```
Comment (mlu-ssoft) [inline libs/map/feature-action-creator-panel/src/lib/content/workflow/function-input.component.tsx:14]: I would this optional

File: libs/map/feature-action-creator-panel/src/lib/content/workflow/workflow.component.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  disabled: true,
+  selected: false,
+};
+
+export const Workflow = () => {
+  const showInputs = false;
+  return (
+    <section className='text-text-primary flex justify-center'>
+      <div>
+        <Node {...args} type='area'>
```
Comment (mlu-ssoft) [inline libs/map/feature-action-creator-panel/src/lib/content/workflow/workflow.component.tsx:19]: I would create new component for each type: AreaComponent, DataSetsComponent, DateRangeComponent, FunctionComponent and here just display those components. Then each component will have its own logic

File: libs/map/feature-action-creator-panel/src/lib/content/workflow/workflow.component.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          {showInputs && <PseudoInput iconName='Polygon' value={'sample aoi'} />}
```
Comment (mlu-ssoft) [inline libs/map/feature-action-creator-panel/src/lib/content/workflow/workflow.component.tsx:20]: can be just string, same in other `PseudoInput` components

**Review by mlu-ssoft — CHANGES_REQUESTED** [review]

File: libs/shared/design-system/src/lib/select/select.tsx (hunk 3 above)
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/select/select.tsx:33]: added comment to do in another ticket

File: libs/map/feature-action-creator-panel/src/lib/content/workflow/node-select.component.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+      { value: 'NDVI', label: t(`${labelPath}.NDVI`) },
+      { value: 'False colour (urban)', label: t(`${labelPath}.FALSE_COLOUR`) },
+      { value: 'Moisture index', label: t(`${labelPath}.MOISTURE_INDEX`) },
+      { value: 'SWIR', label: t(`${labelPath}.SWIR`) },
+      { value: 'NDWI', label: t(`${labelPath}.NDWI`) },
+      { value: 'NDSI', label: t(`${labelPath}.NDSI`) },
+    ];
+  }, [t]);
+
+  const handleChange = (value?: string | null) => {
```
Comment (mlu-ssoft) [inline libs/map/feature-action-creator-panel/src/lib/content/workflow/node-select.component.tsx:25]: add useCallback

File: libs/shared/design-system/src/lib/select/select.tsx [hunk 8]
```diff
@@ -1,76 +1,121 @@
...
-      onChange(selectedOptionValue);
+      onChange(selectedOptionValue.value);
+      setIsOpen(true);
     },
     [onChange]
   );
 
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (ref.current && !(ref.current as Node).contains(event.target as Node)) {
```
Comment (mlu-ssoft) [inline libs/shared/design-system/src/lib/select/select.tsx:54]: maybe there should be different type for useRef?

**Review by mlu-ssoft — CHANGES_REQUESTED** [review]

**Review by mlu-ssoft — APPROVED** [review]

File: libs/shared/design-system/src/lib/select/select.tsx (hunk 8 above)
Comment (KatarzynaPaszko) [reply libs/shared/design-system/src/lib/select/select.tsx:54]: Will be done in another ticket

**Pull request merged**